nuxeo-mongodb-audit/src/main/java/org/nuxeo/mongodb/audit/pageprovider/MongoDBAuditPageProvider.java

```diff
@@ -0,0 +1,347 @@
+            filter = mongoDBBackend.buildFilter(baseQuery, null);
+        } else {
+
+            // Add the quick filters clauses to the fixed part
+            String fixedPart = getFixedPart();
+            if (!StringUtils.isBlank(quickFiltersClause)) {
+                fixedPart = (!StringUtils.isBlank(fixedPart))
+                        ? NXQLQueryBuilder.appendClause(fixedPart, quickFiltersClause) : quickFiltersClause;
+            }
+
+            // Where clause based on DocumentModel
+            String baseQuery = mongoDBBackend.expandQueryVariables(fixedPart, params);
+            filter = buildSearchFilter(baseQuery, whereClause.getPredicates(), getSearchDocumentModel());
+        }
+        return filter;
+    }
+
+    @Override
+    public void refresh() {
+        setCurrentPageOffset(0);
+        super.refresh();
+    }
+
+    @Override
+    public long getResultsCount() {
+        return resultsCount;
+    }
+
+    @Override
+    public List<SortInfo> getSortInfos() {
+        // because ContentView can reuse PageProvider without redefining columns
+        // ensure compat for ContentView configured with JPA log.* sort syntax
+        List<SortInfo> sortInfos = super.getSortInfos();
+        for (SortInfo si : sortInfos) {
+            if (si.getSortColumn().startsWith("log.")) {
+                si.setSortColumn(si.getSortColumn().substring(4));
+            }
+        }
+        return sortInfos;
+    }
+
+    private Bson buildFilter(PredicateDefinition[] predicates, DocumentModel searchDocumentModel) {
+        if (searchDocumentModel == null) {
+            return new Document();
+        }
+
+        List<Bson> filters = new ArrayList<>();
+
+        int nbFilters = 0;
+
+        for (PredicateDefinition predicate : predicates) {
+
+            // extract data from DocumentModel
+            PredicateFieldDefinition[] fieldDef = predicate.getValues();
+            Object[] val = new Object[fieldDef.length];
+            for (int fidx = 0; fidx < fieldDef.length; fidx++) {
+                if (fieldDef[fidx].getXpath() != null) {
+                    val[fidx] = searchDocumentModel.getPropertyValue(fieldDef[fidx].getXpath());
+                } else {
+                    val[fidx] = searchDocumentModel.getProperty(fieldDef[fidx].getSchema(), fieldDef[fidx].getName());
+                }
+                // Convert Calendar objects
+                if (val[fidx] instanceof Calendar) {
+                    val[fidx] = ((Calendar) val[fidx]).getTime();
+                }
+            }
+
+            if (!isNonNullParam(val)) {
+                // skip predicate where all values are null
+                continue;
+            }
+
+            nbFilters++;
+
+            String op = predicate.getOperator();
+            if (op.equalsIgnoreCase("IN")) {
+
+                String[] values = null;
+                if (val[0] instanceof Iterable<?>) {
```

Comment:
Put `val[0]` in a local variable. And are we sure the array cannot be empty? Worth a comment.